src/schema/submissions.ts

```diff
@@ -0,0 +1,113 @@
+import { Post, Submission, User } from './../entity';
+import { IResolvers } from 'graphql-tools';
+import { traceResolvers } from './trace';
+import { Context } from '../Context';
+import { getDiscussionLink, isValidHttpUrl } from '../common';
+import { ValidationError } from 'apollo-server-errors';
+import { GQLEmptyResponse } from './common';
+
+interface GQLArticleSubmission {
+  url: string;
+}
+
+interface GQLSubmissionAvailability {
+  hasAccess: boolean;
+  limit: number;
+  todaySubmissionsCount: number;
+}
+
+export const DEFAULT_SUBMISSION_LIMIT = '3';
+
+export const typeDefs = /* GraphQL */ `
+  type SubmissionAvailability {
+    hasAccess: Boolean!
+    limit: Int!
+    todaySubmissionsCount: Int!
+  }
+
+  extend type Query {
+    """
+    Information regarding the access of submitting community links
+    """
+    submissionsAndLimit: SubmissionAvailability!
+  }
+  extend type Mutation {
+    """
+    Submit an article to surface on users feed
+    """
+    submitArticle(url: String!): EmptyResponse @auth
```

Comment:
We might want to return the submissions object even though the UI will do nothing with it. What do you think?
It might be future proof

Reply:
Can't agree more! Let's do it! :ship